### PR TITLE
Fix some small pathing issues with new script

### LINKED ISSUE #89: New build script break on running in a real component
### Please describe the bug

- `__dirname` is undefined, causing path resolution to break
- readme variant is ignored?
- api table formatting completely chokes and breaks the whole `api.md` file

### Reproducing the error

None

### Expected behavior

_No response_

### What browsers are you seeing the problem on?

_No response_

### Additional context

_No response_

### Exit criteria

_No response_

### PR DESCRIPTION
**Pull request by DukeFerdinand**
# Alaska Airlines Pull Request

Since we're on a breaking change 3.0.0, there are other non-issues that will completely break your docs build, but these three commits were actually problems in the library :)

Closes #89 

## Before Submitting this pull request:
- Link all tickets in this repository related to this PR in the `Development` section
_note: all pull requests require at least one linked ticket_
- If this PR is `Ready For Review`, all ticket's linked under `Development` must have their status changed to `Ready For Review` as well

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I have performed a self-review of my own update.**

## Summary by Sourcery

Fix path handling issues in the fromAuroComponentRoot function and enhance file processing capabilities by adding preProcessors. Update AuroLibraryUtils to use import.meta.url for directory path determination.

Bug Fixes:
- Fix path handling in fromAuroComponentRoot function to correctly process paths starting with a slash.

Enhancements:
- Add preProcessors property to the FileProcessorConfig to allow additional processing on content before template filling.
- Modify getDirname method in AuroLibraryUtils to use import.meta.url for determining the directory path.

## Summary by Sourcery

Fix path handling issues in the fromAuroComponentRoot function and enhance file processing capabilities by adding preProcessors. Update AuroLibraryUtils to use import.meta.url for directory path determination.

Bug Fixes:
- Fix path handling in fromAuroComponentRoot function to correctly process paths starting with a slash.

Enhancements:
- Add preProcessors property to the FileProcessorConfig to allow additional processing on content before template filling.
- Modify getDirname method in AuroLibraryUtils to use import.meta.url for determining the directory path.

### CODE REVIEW

File: scripts/utils/auroLibraryUtils.mjs [hunk 1]
```diff
@@ -12,13 +12,7 @@ import {Logger} from "./logger.mjs";
...
 export default class AuroLibraryUtils {
   getDirname() {
-    if (typeof __dirname === 'undefined') {
-      Logger.warn('Unable to determine project root as __dirname is not defined. Assuming current directory is okay!', true);
-      return '';
-    }
-
-    // eslint-disable-next-line no-undef
-    return __dirname;
+    return import.meta.url.replace("file://", "");
```
[the remark this answers is not in the document]
Comment (DukeFerdinand) [reply scripts/utils/auroLibraryUtils.mjs:15]: can you explain what this does and why I'd want to use this instead?

File: scripts/utils/sharedFileProcessorUtils.mjs [hunk 2]
```diff
@@ -241,6 +244,13 @@ export async function processContentForFile(config) {
   // 3a. Read the output file contents
   let fileContents = await fs.readFile(output, {encoding: 'utf-8'});
 
+  // 3c. Run any post-processors
```
[the remark this answers is not in the document]
Comment (DukeFerdinand) [reply scripts/utils/sharedFileProcessorUtils.mjs]: wow good catch. can't believe I missed this :(

File: scripts/utils/auroLibraryUtils.mjs (hunk 1 above)
[the remark this answers is not in the document]
Comment (DukeFerdinand) [reply scripts/utils/auroLibraryUtils.mjs]: I decided to just consume this since it seems to do what I want in a more "correct" way

**Comment by DukeFerdinand** [comment]
@sourcery-ai review

File: scripts/utils/sharedFileProcessorUtils.mjs [hunk 3]
```diff
@@ -241,17 +244,24 @@ export async function processContentForFile(config) {
   // 3a. Read the output file contents
   let fileContents = await fs.readFile(output, {encoding: 'utf-8'});
 
-  // 3b. Replace template variables in output file
+  // 3b. Run any pre-processors
+  if (config.preProcessors) {
+    for (const processor of config.postProcessors) {
```
[the remark this answers is not in the document]
Comment (DukeFerdinand) [reply scripts/utils/sharedFileProcessorUtils.mjs]: fixed in next commit

File: scripts/utils/sharedFileProcessorUtils.mjs [hunk 4]
```diff
@@ -75,8 +73,12 @@ export const nonEsmComponents = ['combobox', 'datepicker', 'menu', 'pane', 'sele
  */
 // TODO: test this in auro-flight before merging to main
 export function fromAuroComponentRoot(pathLike) {
-  const currentDir = fileURLToPath(new URL('.', import.meta.url))
-  return path.join(currentDir, `${auroLibraryUtils.getProjectRootPath}${pathLike}`)
+  if (pathLike.startsWith('/')) {
+    // remove the first slash
+    return auroLibraryUtils.getProjectRootPath + pathLike.slice(1)
```
[the remark this answers is not in the document]
Comment (DukeFerdinand) [reply scripts/utils/sharedFileProcessorUtils.mjs]: fixed in next commit

**Comment by DukeFerdinand** [comment]
@sourcery-ai review

**Review by sun-mota — CHANGES_REQUESTED** [review]
I'm getting some error prints.
`npm run build:docs` on Auro-flight
image

**Comment by DukeFerdinand** [comment]
> I'm getting some error prints. `npm run build:docs` on Auro-flight image

This is expected :) it's the main "breaking change" when adopting this new build script

**Review by sun-mota — APPROVED** [review]
confirmed this works with a change(https://github.com/AlaskaAirlines/auro-flight/pull/140)

**Pull request merged**